### ISSUE TITLE
[BUG] Panic and WASM Plugin Compatibility Issues with Next.js 15

### ISSUE THREAD

**Issue by wongmjane** [Dwlad90/stylex-swc-plugin#120]
Hi! I took your `@stylexswc/next-swc-example` and upgraded the Next.js to v15, and it panics and crashes.

The error suggests the `swc_core` might need to be upgraded in order for this to work with Next.js v15

Repro repo: https://github.com/wongmjane/stylexswc-next-swc-example

Error output:
```
➜  nextjs-swc-example git:(main) pnpm dev

> @stylexswc/next-swc-example@0.4.4-rc.1 predev [redacted]/nextjs-swc-example
> pnpm run clean


> @stylexswc/next-swc-example@0.4.4-rc.1 clean [redacted]/nextjs-swc-example
> rimraf .next


> @stylexswc/next-swc-example@0.4.4-rc.1 dev [redacted]/nextjs-swc-example
> next dev

   ▲ Next.js 15.0.3
   - Local:        http://localhost:3000
   - Experiments (use with caution):
     · swcPlugins

 ✓ Starting...
 ⚠ Invalid next.config.js options detected:
 ⚠     Unrecognized key(s) in object: 'swcMinify'
 ⚠ See more info here: https://nextjs.org/docs/messages/invalid-next-config
!!!GETTING WEBPACK CONFIG!!!
======================
Count: 1
Build ID: development
Server: true
Env: dev
!!!GETTING WEBPACK CONFIG!!!
======================
Count: 2
Build ID: development
Server: true
Env: dev
!!!GETTING WEBPACK CONFIG!!!
======================
Count: 3
Build ID: development
Server: false
Env: dev
 ✓ Ready in 870ms
thread 'thread '<unnamed><unnamed>' panicked at ' panicked at /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rkyv-0.7.44/src/impls/core/mod.rs/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rkyv-0.7.44/src/impls/core/mod.rs::266266::6767:
:
called `Result::unwrap()` on an `Err` value: LayoutError
called `Result::unwrap()` on an `Err` value: LayoutErrornote: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
FATAL: An unexpected Turbopack error occurred. Please report the content of /var/folders/nq/452h5tp1229cyy7xhw9wqzkm0000gn/T/next-panic-6f6235e9cdc0be76f20f2c1d10ecd420.log, along with a description of what you were doing when the error occurred, to https://github.com/vercel/next.js/issues/new
thread '<unnamed>' panicked at /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rkyv-0.7.44/src/impls/core/mod.rs:266:67:
thread 'called `Result::unwrap()` on an `Err` value: LayoutError<unnamed>' panicked at
/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rkyv-0.7.44/src/impls/core/mod.rsnote: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
:266:67:
called `Result::unwrap()` on an `Err` value: LayoutError
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread '<unnamed>' panicked at alloc/src/raw_vec.rs:24:5:
capacity overflow
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread '<unnamed>' panicked at /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rkyv-0.7.44/src/impls/core/mod.rs:266:67thread ':
<unnamed>called `Result::unwrap()` on an `Err` value: LayoutError' panicked at
/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rkyv-0.7.44/src/impls/core/mod.rs:note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
266:67:
called `Result::unwrap()` on an `Err` value: LayoutError
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread '<unnamed>' panicked at /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rkyv-0.7.44/src/impls/core/mod.rs:266:67:
called `Result::unwrap()` on an `Err` value: LayoutError
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
 ○ Compiling / ...
FATAL: An unexpected Turbopack error occurred. Please report the content of /var/folders/nq/452h5tp1229cyy7xhw9wqzkm0000gn/T/next-panic-6f6235e9cdc0be76f20f2c1d10ecd420.log, along with a description of what you were doing when the error occurred, to https://github.com/vercel/next.js/issues/new
 ⨯ ./node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/client/app-next-dev.js
Error: failed to process failed to invoke plugin: failed to invoke plugin on 'Some("[redacted]/nextjs-swc-example/node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/client/app-next-dev.js")'

Caused by:
    wasm plugin bytecheck failed "check bytes error: check failed for tuple struct member 0: check failed for enum tuple variant Module: check failed for tuple struct member 0: check failed for struct member body: check failed for slice index 0: check failed for enum tuple variant Stmt: check failed for tuple struct member 0: check failed for enum tuple variant Expr: check failed for tuple struct member 0: check failed for struct member expr: pointer out of bounds: base 0x54c3e071268 offset -5728 not in range 0x54c3e071200..0x54c3e0713cc"

Stack backtrace:
   0: _napi_register_module_v1
   1: _napi_register_module_v1
   2: _BrotliDecoderVersion
   3: _BrotliDecoderVersion
   4: _BrotliDecoderVersion
   5: _BrotliDecoderVersion
   6: _BrotliDecoderVersion
   7: _BrotliDecoderVersion
   8: _BrotliDecoderVersion
   9: _BrotliDecoderVersion
  10: _BrotliDecoderVersion
  11: <unknown>
  12: <unknown>
  13: _BrotliDecoderVersion
  14: _BrotliDecoderVersion
  15: _BrotliDecoderVersion
  16: _BrotliDecoderVersion
  17: _BrotliDecoderVersion
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>

Stack backtrace:
   0: _napi_register_module_v1
   1: _napi_register_module_v1
   2: <unknown>
   3: <unknown>
thread '<unnamed>' panicked at /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rkyv-0.7.44/src/impls/core/mod.rsthread ':<unnamed>266:' panicked at 67/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rkyv-0.7.44/src/impls/core/mod.rs:
:called `Result::unwrap()` on an `Err` value: LayoutError266
:note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
67:
called `Result::unwrap()` on an `Err` value: LayoutError
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread '<unnamed>' panicked at /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rkyv-0.7.44/src/impls/core/mod.rs:266:67:
called `Result::unwrap()` on an `Err` value: LayoutError
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread '<unnamed>' panicked at /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rkyv-0.7.44/src/impls/core/mod.rs:266:67:
called `Result::unwrap()` on an `Err` value: LayoutError
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread '<unnamed>' panicked at alloc/src/raw_vec.rs:24:5:
capacity overflow
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'thread '<unnamed>thread '<unnamed><unnamed>' panicked at ' panicked at ' panicked at /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rkyv-0.7.44/src/impls/core/mod.rs/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rkyv-0.7.44/src/impls/core/mod.rs:/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rkyv-0.7.44/src/impls/core/mod.rs:266:266::26667:67:
67:
called `Result::unwrap()` on an `Err` value: LayoutErrorcalled `Result::unwrap()` on an `Err` value: LayoutError:
called `Result::unwrap()` on an `Err` value: LayoutError

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
FATAL: An unexpected Turbopack error occurred. Please report the content of /var/folders/nq/452h5tp1229cyy7xhw9wqzkm0000gn/T/next-panic-6f6235e9cdc0be76f20f2c1d10ecd420.log, along with a description of what you were doing when the error occurred, to https://github.com/vercel/next.js/issues/new
 ⨯ ./node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/client/app-next-dev.js
Error: failed to process failed to invoke plugin: failed to invoke plugin on 'Some("[redacted]/nextjs-swc-example/node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/client/app-next-dev.js")'

Caused by:
    wasm plugin bytecheck failed "check bytes error: check failed for tuple struct member 0: check failed for enum tuple variant Module: check failed for tuple struct member 0: check failed for struct member body: check failed for slice index 0: check failed for enum tuple variant Stmt: check failed for tuple struct member 0: check failed for enum tuple variant Expr: check failed for tuple struct member 0: check failed for struct member expr: pointer out of bounds: base 0x54c3e071868 offset -5728 not in range 0x54c3e071800..0x54c3e0719cc"

Stack backtrace:
   0: _napi_register_module_v1
   1: _napi_register_module_v1
   2: _BrotliDecoderVersion
   3: _BrotliDecoderVersion
   4: _BrotliDecoderVersion
   5: _BrotliDecoderVersion
   6: _BrotliDecoderVersion
   7: _BrotliDecoderVersion
   8: _BrotliDecoderVersion
   9: _BrotliDecoderVersion
  10: _BrotliDecoderVersion
  11: <unknown>
  12: <unknown>
  13: _BrotliDecoderVersion
  14: _BrotliDecoderVersion
  15: _BrotliDecoderVersion
  16: _BrotliDecoderVersion
  17: _BrotliDecoderVersion
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>

Stack backtrace:
   0: _napi_register_module_v1
   1: _napi_register_module_v1
   2: <unknown>
   3: <unknown>
!!!GETTING WEBPACK CONFIG!!!
======================
Count: 4
Build ID: development
Server: false
Env: dev
thread '<unnamed>' panicked at alloc/src/raw_vec.rs:24:5:
capacity overflow
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread '<unnamed>' panicked at alloc/src/raw_vec.rs:24:5:
capacity overflow
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
FATAL: An unexpected Turbopack error occurred. Please report the content of /var/folders/nq/452h5tp1229cyy7xhw9wqzkm0000gn/T/next-panic-6f6235e9cdc0be76f20f2c1d10ecd420.log, along with a description of what you were doing when the error occurred, to https://github.com/vercel/next.js/issues/new
thread 'thread '<unnamed><unnamed>' panicked at ' panicked at /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rkyv-0.7.44/src/impls/core/mod.rs/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rkyv-0.7.44/src/impls/core/mod.rs::266266::6767:
:
called `Result::unwrap()` on an `Err` value: LayoutErrorcalled `Result::unwrap()` on an `Err` value: LayoutError

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread '<unnamed>' panicked at thread '<unnamed>/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rkyv-0.7.44/src/impls/core/mod.rs' panicked at :/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rkyv-0.7.44/src/impls/core/mod.rs266:266::67:
67:
called `Result::unwrap()` on an `Err` value: LayoutErrorcalled `Result::unwrap()` on an `Err` value: LayoutError

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread '<unnamed>' panicked at alloc/src/raw_vec.rs:24:5:
capacity overflow
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'thread '<unnamed><unnamed>' panicked at ' panicked at /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rkyv-0.7.44/src/impls/core/mod.rs/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rkyv-0.7.44/src/impls/core/mod.rs:266:266thread ':<unnamed>:67' panicked at 67:
/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rkyv-0.7.44/src/impls/core/mod.rscalled `Result::unwrap()` on an `Err` value: LayoutError:
:
called `Result::unwrap()` on an `Err` value: LayoutErrornote: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
266
:note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
67:
called `Result::unwrap()` on an `Err` value: LayoutError
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
 ⨯ ./node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/client/next-dev.js
Error: failed to process failed to invoke plugin: failed to invoke plugin on 'Some("[redacted]/nextjs-swc-example/node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/client/next-dev.js")'

Caused by:
    0: failed to invoke `[redacted]/nextjs-swc-example/node_modules/.pnpm/@stylexswc+swc-plugin@0.4.4-rc.1/node_modules/@stylexswc/swc-plugin/dist/stylex_swc_plugin.wasm` as js transform plugin at [redacted]/nextjs-swc-example/node_modules/.pnpm/@stylexswc+swc-plugin@0.4.4-rc.1/node_modules/@stylexswc/swc-plugin/dist/stylex_swc_plugin.wasm
    1: failed to run Wasm plugin transform. Please ensure the version of `swc_core` used by the plugin is compatible with the host runtime. See the documentation for compatibility information. If you are an author of the plugin, please update `swc_core` to the compatible version.

                       Note that if you want to use the os features like filesystem, you need to use `wasi`. Wasm itself does not have concept of filesystem.

                       https://swc.rs/docs/plugin/selecting-swc-core

                       See https://plugins.swc.rs/versions/from-plugin-runner/4.0.0 for the list of the compatible versions.

                       Build info:
                           Date: 2024-11-07
                           Timestamp: 2024-11-07T09:53:12.683840000Z

                       Version info:
                           swc_plugin_runner: 4.0.0
                           Dependencies: anyhow 1.0.93,codspeed-criterion-compat 2.7.2,criterion 0.5.1,enumset 1.1.5,futures 0.3.31,once_cell 1.20.2,parking_lot 0.12.3,serde 1.0.214,serde_json 1.0.132,swc_atoms 2.0.0,swc_common 4.0.0,swc_css_ast 4.0.1,swc_css_parser 4.0.0,swc_ecma_ast 4.0.1,swc_ecma_loader 4.0.0,swc_ecma_parser 5.0.0,swc_ecma_visit 4.0.0,swc_malloc 1.0.0,swc_plugin_proxy 4.0.0,testing 4.0.0,tokio 1.41.0,tracing 0.1.40,vergen 9.0.1,virtual-fs 0.16.0,wasmer 4.3.7,wasmer-cache 4.3.7,wasmer-compiler-cranelift 4.3.7,wasmer-wasix 0.27.0

    2: RuntimeError: unreachable

Stack backtrace:
   0: _napi_register_module_v1
   1: _BrotliDecoderVersion
   2: _BrotliDecoderVersion
   3: _BrotliDecoderVersion
   4: _BrotliDecoderVersion
   5: _BrotliDecoderVersion
   6: _BrotliDecoderVersion
   7: _BrotliDecoderVersion
   8: _BrotliDecoderVersion
   9: _BrotliDecoderVersion
  10: _BrotliDecoderVersion
  11: <unknown>
  12: <unknown>
  13: _BrotliDecoderVersion
  14: _BrotliDecoderVersion
  15: _BrotliDecoderVersion
  16: _BrotliDecoderVersion
  17: _BrotliDecoderVersion
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>

Stack backtrace:
   0: _napi_register_module_v1
   1: _napi_register_module_v1
   2: <unknown>
   3: <unknown>
 ⨯ Error: ENOENT: no such file or directory, open '[redacted]/nextjs-swc-example/.next/fallback-build-manifest.json'
    at path (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/load-manifest.ts:44:41)
    at manifestPath (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/load-components.ts:82:29)
    at async loadDefaultErrorComponentsImpl (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/load-default-error-components.ts:66:20)
    at async DevServer.getFallbackErrorComponents (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/dev/next-dev-server.ts:901:11)
    at async DevServer.renderErrorToResponseImpl (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/base-server.ts:4003:33)
    at async DevServer.pipeImpl (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/base-server.ts:1698:20)
    at async NextNodeServer.handleCatchallRenderRequest (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/next-server.ts:972:8)
    at async DevServer.handleRequestImpl (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/base-server.ts:1462:8)
    at async (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/dev/next-dev-server.ts:514:13)
    at async Span.traceAsyncFn (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/trace/trace.ts:143:13)
    at async DevServer.handleRequest (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/dev/next-dev-server.ts:512:19)
    at async invokeRender (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/lib/router-server.ts:284:10)
    at async handleRequest (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/lib/router-server.ts:530:15)
    at async requestHandlerImpl (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/lib/router-server.ts:576:6)
    at async Server.requestListener (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/lib/start-server.ts:146:6) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '[redacted]/nextjs-swc-example/.next/fallback-build-manifest.json'
}
Error: ENOENT: no such file or directory, open '[redacted]/nextjs-swc-example/.next/fallback-build-manifest.json'
    at path (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/load-manifest.ts:44:41)
    at manifestPath (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/load-components.ts:82:29)
    at async loadDefaultErrorComponentsImpl (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/load-default-error-components.ts:66:20)
    at async DevServer.getFallbackErrorComponents (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/dev/next-dev-server.ts:901:11)
    at async DevServer.renderErrorToResponseImpl (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/base-server.ts:4003:33)
    at async DevServer.pipeImpl (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/base-server.ts:1698:20)
    at async NextNodeServer.handleCatchallRenderRequest (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/next-server.ts:972:8)
    at async DevServer.handleRequestImpl (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/base-server.ts:1462:8)
    at async (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/dev/next-dev-server.ts:514:13)
    at async Span.traceAsyncFn (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/trace/trace.ts:143:13)
    at async DevServer.handleRequest (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/dev/next-dev-server.ts:512:19)
    at async invokeRender (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/lib/router-server.ts:284:10)
    at async handleRequest (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/lib/router-server.ts:530:15)
    at async requestHandlerImpl (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/lib/router-server.ts:576:6)
    at async Server.requestListener (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/lib/start-server.ts:146:6) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '[redacted]/nextjs-swc-example/.next/fallback-build-manifest.json'
}
Error: ENOENT: no such file or directory, open '[redacted]/nextjs-swc-example/.next/fallback-build-manifest.json'
    at path (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/load-manifest.ts:44:41)
    at manifestPath (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/load-components.ts:82:29)
    at async loadDefaultErrorComponentsImpl (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/load-default-error-components.ts:66:20)
    at async DevServer.getFallbackErrorComponents (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/dev/next-dev-server.ts:901:11)
    at async DevServer.renderErrorToResponseImpl (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/base-server.ts:4003:33)
    at async pipe.req.req (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/base-server.ts:3818:25)
    at async DevServer.pipeImpl (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/base-server.ts:1698:20)
    at async (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/dev/next-dev-server.ts:514:13)
    at async Span.traceAsyncFn (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/trace/trace.ts:143:13)
    at async DevServer.handleRequest (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/dev/next-dev-server.ts:512:19)
    at async invokeRender (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/lib/router-server.ts:284:10)
    at async requestHandlerImpl (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/lib/router-server.ts:589:15)
    at async Server.requestListener (node_modules/.pnpm/next@15.0.3_@babel+core@7.26.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/src/server/lib/start-server.ts:146:6) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '[redacted]/nextjs-swc-example/.next/fallback-build-manifest.json'
}
```

Thanks!

**Comment by Dwlad90** [comment]
hi @wongmjane,
`next@15` is really not supported by the plugin, in the future support will be added when most projects switch to version 15, since the SWC plugin will not have backward compatibility with `nextjs@14`.

In any case, I recommend using `@stylexswc/nextjs-plugin` with `@stylexswc/rs-compiler` instead of SWC, since it is the main package that uses the SWC plugin code under the hood, and at the same time gives an API similar to babel plugin, ease of use with various bundlers and does not depend on SWC version.

PS,
I checked your reproduction with `@stylexswc/rs-compiler` and found a bug in path resolution with using `pnpm`. I will try to fix it as soon as possible. With `npm` everything works fine together with `next@15`

**Comment by Dwlad90** [comment]
The problem with paths resolving of packages installed via `pnpm` was fixed in version `0.4.4-rc.3`

**Comment by wongmjane** [comment]
Closing this as the SWC Plugin has been explicitly marked as deprecated in a71ade7b7c8810a6360e458ca1c412a55e49b2fe
